### PR TITLE
Add action to message discord on PRs

### PR DESCRIPTION
**Pull request by ShawnCollinge**
Author: Shawn

## What changes were made?
Adds action to message the github channel on discord when a PR gets created/merged/closed.

## Why are these changes important/necessary?
Helps us keep track of everything.

### CODE REVIEW

File: .github/workflows/message-discord.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Send Discord notification
+        env:
+          DISCORD_WEBHOOK_URL: ${{ secrets.DISCORD_WEBHOOK_URL }}
+        run: |
+          # Construct the message based on the type of PR event
+          if [[ "${{ github.event.action }}" == "opened" ]]; then
+            MESSAGE="📢 A new pull request has been opened in \`${{ github.repository }}\` by \`${{ github.actor }}\`.\n🔗 [PR #${{ github.event.number }} - ${{ github.event.pull_request.title }}](${{ github.event.pull_request.html_url }})";
```
Comment (elimelt) [inline .github/workflows/message-discord.yaml:18]: nit: line breaks to make it more readable, but this isn't a big deal

e.g.

```bash
MESSAGE="📢 A new pull request has been opened in \`${{ github.repository }}\`"
MESSAGE+="by \`${{ github.actor }}\`.\n"
MESSAGE+="🔗 [PR #${{ github.event.number }} - ${{ github.event.pull_request.title }}](${{ github.event.pull_request.html_url }})";"
```

Comment (elimelt) [reply .github/workflows/message-discord.yaml:18]: Also though, you can do single quotes and just rely on the actual line break instead of using a `\n`

**Review by elimelt — APPROVED** [review]
LGTM besides nit

File: .github/workflows/message-discord.yaml (hunk 1 above)
Comment (ShawnCollinge) [reply .github/workflows/message-discord.yaml]: 10-4

**Pull request merged**